great_expectations/self_check/util.py

```diff
@@ -99,60 +99,82 @@
-    # with https://github.com/googleapis/python-bigquery-sqlalchemy/pull/251. If that
-    # fix isn't present, add this "dialect" attribute here:
-    if not hasattr(pybigquery.sqlalchemy_bigquery, "dialect"):
-        pybigquery.sqlalchemy_bigquery.dialect = (
-            pybigquery.sqlalchemy_bigquery.BigQueryDialect
-        )
+    import sqlalchemy_bigquery.sqlalchemy_bigquery as sqla_bigquery
+    import sqlalchemy_bigquery.sqlalchemy_bigquery as BigQueryDialect
 
-    # Sometimes "pybigquery.sqlalchemy_bigquery" fails to self-register in Azure (our CI/CD pipeline) in certain cases, so we do it explicitly.
-    # (see https://stackoverflow.com/questions/53284762/nosuchmoduleerror-cant-load-plugin-sqlalchemy-dialectssnowflake)
-    sqlalchemy.dialects.registry.register(
-        "bigquery", "pybigquery.sqlalchemy_bigquery", "dialect"
-    )
+    sqlalchemy.dialects.registry.register("bigquery", _BIGQUERY_MODULE_NAME, "dialect")
+    bigquery_types_tuple = None
+    BIGQUERY_TYPES = {
+        "INTEGER": sqla_bigquery.INTEGER,
+        "NUMERIC": sqla_bigquery.NUMERIC,
+        "STRING": sqla_bigquery.STRING,
+        "BIGNUMERIC": sqla_bigquery.BIGNUMERIC,
+        "BYTES": sqla_bigquery.BYTES,
+        "BOOL": sqla_bigquery.BOOL,
+        "BOOLEAN": sqla_bigquery.BOOLEAN,
+        "TIMESTAMP": sqla_bigquery.TIMESTAMP,
+        "TIME": sqla_bigquery.TIME,
+        "FLOAT": sqla_bigquery.FLOAT,
+        "DATE": sqla_bigquery.DATE,
+        "DATETIME": sqla_bigquery.DATETIME,
+    }
+except ImportError:
     try:
-        getattr(pybigquery.sqlalchemy_bigquery, "INTEGER")
-        bigquery_types_tuple = {}
-        BIGQUERY_TYPES = {
-            "INTEGER": pybigquery.sqlalchemy_bigquery.INTEGER,
-            "NUMERIC": pybigquery.sqlalchemy_bigquery.NUMERIC,
-            "STRING": pybigquery.sqlalchemy_bigquery.STRING,
-            "BIGNUMERIC": pybigquery.sqlalchemy_bigquery.BIGNUMERIC,
-            "BYTES": pybigquery.sqlalchemy_bigquery.BYTES,
-            "BOOL": pybigquery.sqlalchemy_bigquery.BOOL,
-            "BOOLEAN": pybigquery.sqlalchemy_bigquery.BOOLEAN,
-            "TIMESTAMP": pybigquery.sqlalchemy_bigquery.TIMESTAMP,
-            "TIME": pybigquery.sqlalchemy_bigquery.TIME,
-            "FLOAT": pybigquery.sqlalchemy_bigquery.FLOAT,
-            "DATE": pybigquery.sqlalchemy_bigquery.DATE,
-            "DATETIME": pybigquery.sqlalchemy_bigquery.DATETIME,
-        }
-    except AttributeError:
-        # In older versions of the pybigquery driver, types were not exported, so we use a hack
-        logger.warning(
-            "Old pybigquery driver version detected. Consider upgrading to 0.4.14 or later."
-        )
-        from collections import namedtuple
+        import pybigquery.sqlalchemy_bigquery as sqla_bigquery
+        import pybigquery.sqlalchemy_bigquery as BigQueryDialect
 
-        BigQueryTypes = namedtuple(
-            "BigQueryTypes", sorted(pybigquery.sqlalchemy_bigquery._type_map)
+        logger.warn(
```

Comment:
Please make this a deprecation warning, ex:
`warnings.warn("The pybigquery package is obsolete, please use sqlalchemy-bigquery", DeprecationWarning)`